src/util/wallet-utils/src/callContract.ts

```diff
@@ -0,0 +1,51 @@
+import { request } from "sats-connect";
+
+import { StacksNetwork, StacksNetworkName } from "@stacks/network";
+import {
+  getFordefiBTCProviderOrThrow,
+  getLeatherBTCProviderOrThrow,
+} from "./util/btc-provider";
+
+type RequestArgs = {
+  txHex: string;
+  network?: StacksNetworkName | StacksNetwork;
+};
+
+type CallContract = (params: RequestArgs) => Promise<string>;
+
+export const callContractXverse: CallContract = async (params) => {
+  const response = await request("stx_signTransaction", {
+    transaction: params.txHex,
+  });
+  if (response.status === "error") {
+    throw new Error(response.error.message);
+  }
+  return response.result.transaction;
+};
+
+export const callContractLeather: CallContract = async (params) => {
```

Comment:
I tried this one (in devenv) and it worked fine, did you try `callContractXverse` and `callContractFordefi`?

Reply:
I tested xverse I think it's prime time we mergge the connect pr

Reply:
Tried xverse not sure I can try fordefi in devenv

Reply:
I can just mark it as unsupported for now

Reply:
done!